OPHD/UI/Core/RadioButtonGroup.cpp

```diff
@@ -0,0 +1,78 @@
+#include "RadioButtonGroup.h"
+
+#include "../../Cache.h"
+#include "../../Constants.h"
+
+#include <NAS2D/Utility.h>
+#include <NAS2D/Renderer/Renderer.h>
+#include <NAS2D/MathUtils.h>
+
+#include <algorithm>
+
+
+using namespace NAS2D;
+
+
+RadioButtonGroup::RadioButtonGroup(std::vector<ButtonInfo> buttonInfos)
+{
+	mRadioButtons.reserve(buttonInfos.size());
+	for(auto &buttonInfo : buttonInfos)
+	{
+		NAS2D::Vector<int> offset = {0, 13};
+		offset.y = mRadioButtons.size() * offset.y;
+
+		auto &button = mRadioButtons.emplace_back(buttonInfo.name, this, buttonInfo.delegate);
+		button.visible(visible());
+		button.position(mRect.startPoint() + offset);
+	}
+}
+
+void RadioButtonGroup::onMove(NAS2D::Vector<int> displacement)
+{
+	Control::onMove(displacement);
+
+	for (auto &control : mRadioButtons)
+	{
+		control.position(control.position() + displacement);
+	}
+}
+
+void RadioButtonGroup::update()
+{
+	if (!visible()) { return; }
+
+	if( mRadioButtons.size() > 0 && !hasOneSelectedItem() ) {
+		clearSelection();
+		mRadioButtons[0].click();
+	}
+
+	for (auto &control : mRadioButtons)
+	{
+		control.update();
+	}
+}
+
+void RadioButtonGroup::clearSelection()
+{
+	if (!visible()) { return; }
+
+	for (auto &control : mRadioButtons)
+	{
+		control.checked(false);
+	}
+}
+
+bool RadioButtonGroup::hasOneSelectedItem()
+{
+	int count = 0;
+	for (auto &control : mRadioButtons)
+	{
+		if(control.checked()) { count++; }
+	}
+	return count == 1;
+}
```

Comment:
I would really like for this method to not be needed.

Reply:
Okay, I have removed it locally for now, still testing if this assertion can be made to hold without it.

Reply:
Removed in latest push.